InformationScripting/src/queries/AstQuery.cpp

```diff
@@ -67,37 +69,32 @@ TupleSet AstQuery::executeLinear(TupleSet input)
 
 void AstQuery::registerDefaultQueries()
 {
-	auto& registry = QueryRegistry::instance();
-	registry.registerQueryConstructor("classes", [](Model::Node* target, QStringList args) {
+	QueryRegistry::instance().registerQueryConstructor("classes", [](Model::Node* target, QStringList args) {
 		setTypeTo(args, "Class");
 		return new AstQuery(&AstQuery::genericQuery, target, args);
 	});
-	registry.registerQueryConstructor("methods", [](Model::Node* target, QStringList args) {
+	QueryRegistry::instance().registerQueryConstructor("methods", [](Model::Node* target, QStringList args) {
 		setTypeTo(args, "Method");
 		return new AstQuery(&AstQuery::genericQuery, target, args);
 	});
-	registry.registerQueryConstructor("bases", [](Model::Node* target, QStringList args) {
-		return new AstQuery(&AstQuery::baseClassesQuery, target, args);
-	});
-	registry.registerQueryConstructor("toClass", [](Model::Node* target, QStringList args) {
+	QueryRegistry::instance().registerQueryConstructor("toClass", [](Model::Node* target, QStringList args) {
 		setTypeTo(args, "Class");
 		return new AstQuery(&AstQuery::toParentType, target, args);
 	});
-	registry.registerQueryConstructor("callgraph", [](Model::Node* target, QStringList args) {
-		return new AstQuery(&AstQuery::callGraph, target, args);
-	});
-	registry.registerQueryConstructor("ast", [](Model::Node* target, QStringList args) {
-		return new AstQuery(&AstQuery::genericQuery, target, args);
-	});
-	registry.registerQueryConstructor("toParent", [](Model::Node* target, QStringList args) {
-		return new AstQuery(&AstQuery::toParentType, target, args);
-	});
-	registry.registerQueryConstructor("uses", [](Model::Node* target, QStringList args) {
-		return new AstQuery(&AstQuery::usesQuery, target, args);
-	});
-	registry.registerQueryConstructor("type", [](Model::Node* target, QStringList args) {
-		return new AstQuery(&AstQuery::typeFilter, target, args);
-	});
+
+	auto registerQuery = [](const QString& name, auto methodToCall) {
```

Comment:
I would put this lambda on top, and add another string argument, which if not null will be used in a call to `setTypeTo(args, argument)`. This way you can do all of the registrations using the lambda.

Since lambdas cannot have default arguments, you can consider also making a private static function instead and settings the last argument to be a null QString by default.